src/utils/mapNameValueFields.js

```diff
@@ -1,21 +1,26 @@
-export const mapNameValueFields = (data, otherThreshold = 0.04) => {
+import { DEFAULT_OTHER_THRESHOLD_PERCENTAGE } from "../constants";
+
+export const mapNameValueFields = (data, otherThreshold = DEFAULT_OTHER_THRESHOLD_PERCENTAGE) => {
     if (!data)
         return [];
 
     // Accumulate all values to compute on them later
     const sumOfAllValues = Object.values(data).reduce((acc, v) => acc + v, 0);
+    const numCategoriesBelowThreshold = Object.values(data).filter(
+        (val) => val / sumOfAllValues < otherThreshold
```

Comment:
I'll attach most of the comments on `mapNameValueFields() `here: 

This is legacy code that has been built on top of a few times (including now!) that could probably benefit from a rewrite. The new code indeed loops twice through every array, but these are always small arrays; the alternative is a bunch of extra conditional handling inside the main loop. Again, a total rewrite could make this cleaner. 

Other issues are also a function of incremental changes elsewhere in the code. For example, `mapNameValueFields` is used only for pie charts, which don't even show categories with zero members, they are filtered out. So that could probably be done here instead. Similarly, a threshold parameter is always passed, so there's no real reason for the default assignment.